src/model/behavioral/simulation.py

```diff
@@ -124,6 +130,17 @@ def step(self,step_length,logger):
 		# epidemicon
 		##############################################################################
 
+
+	def get_agents_location(self):
+		#record the agents location in every steps
+		agent_location = []
+		for agent in self.agents:
+			agent_location.append([agent.agent_id, agent.coordinate.get_lat_lon()])
+		# dataframe = pd.DataFrame(data=agent_location,columns=["id","coordinate"])
+		dataframe = pd.DataFrame(data=agent_location)
+		dataframe.to_csv("location.csv", mode='a', index=False, sep=",", header=False)
```

Comment:
If you generate a data frame save the location to a file every step, the simulation will be VERY slow. Also, it is not necessary to save to a DataFrame if you are going to save to a file immediately.

It is better if you cache the saved locations, and save them to a file occasionally.

I believe that the `logger` library allows you to do that. Please check that